### PR TITLE
Processed the field ( paid to provider) when the update

### PR DESCRIPTION
**Pull request by habibdevsy**
- Processed the field ( paid to provider) when the update 
- Prevent the captain from modifying a field (paidToProvider) after adding a paid to the admin

### CODE REVIEW

File: backend/src/Constant/Order/OrderAmountCashConstant.php [hunk 1]
```diff
@@ -9,4 +9,8 @@ final class OrderAmountCashConstant
     const ORDER_PAID_FLAG_NO = 2;
 
     const NOT_ORDER_CASH = "error not orders cash";
+    
+    const CAPTAIN_NOT_ALLOWED_TO_EDIT_ORDER_PAID_FLAG = false;
+
+    const CAPTAIN_NOT_ALLOWED_TO_EDIT_ORDER_PAID_FLAG_STRING = "not allowed update";
```
Comment (ralsalloum) [inline backend/src/Constant/Order/OrderAmountCashConstant.php]: why not integer value?

File: backend/src/Constant/Order/OrderAmountCashConstant.php [hunk 2]
```diff
@@ -9,4 +9,8 @@ final class OrderAmountCashConstant
...
 }
```
Comment (ralsalloum) [inline backend/src/Constant/Order/OrderAmountCashConstant.php]: empty line at the end of the file

File: backend/src/Service/CaptainAmountFromOrderCash/CaptainAmountFromOrderCashService.php [hunk 3]
```diff
@@ -61,4 +67,16 @@ public function getCashOrdersPaymentsByCaptainId(int $captainId): array
     {
         return $this->captainAmountFromOrderCashManager->getCashOrdersPaymentsByCaptainId($captainId);
     }
+
+    // Is the captain allowed to edit?
+    public function getEditingByCaptain(int $orderId): ?bool
+    {
+        $captainAmountFromOrderCash = $this->captainAmountFromOrderCashManager->getCaptainAmountFromOrderCashByOrderId($orderId);
+     
+        if($captainAmountFromOrderCash === null) {
```
Comment (ralsalloum) [inline backend/src/Service/CaptainAmountFromOrderCash/CaptainAmountFromOrderCashService.php]: Avoid using null in conditional statements, and as a set value.

Comment (habibdevsy) [reply backend/src/Service/CaptainAmountFromOrderCash/CaptainAmountFromOrderCashService.php]: done

**Comment by habibdevsy** [comment]
done all

**Review by ralsalloum — APPROVED** [review]

**Review by ralsalloum — APPROVED** [review]

**Pull request merged**